service/history/workflow/mutable_state_impl.go

```diff
@@ -365,7 +366,9 @@ func NewSanitizedMutableState(
 	}
 
 	// sanitize data
-	mutableState.executionInfo.LastFirstEventTxnId = common.EmptyVersion
+	mutableState.executionInfo.LastFirstEventTxnId = lastFirstEventTxnID
+	mutableState.executionInfo.CloseVisibilityTaskId = common.EmptyVersion
+	mutableState.executionInfo.CloseTransferTaskId = common.EmptyVersion
```

Comment:
Shall we sanitize `LastEventTaskId` field as well?

Reply:
No. This field is to calculate the vector clock. It should not be changed.